src/TransitionSwitch.js

```diff
@@ -32,7 +32,7 @@ export class TransitionSwitch extends React.Component {
 
     static contextTypes = {
         router: PropTypes.shape({
-            route: PropTypes.object.isRequired

```

Comment:
Good catch, but we also need the route actually, if you can add it back I'd merge the PR.